sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/ScrubEtagPolicy.java

```diff
@@ -49,15 +46,18 @@ private HttpResponse scrubETagHeader(HttpResponse unprocessedResponse) {
         String eTag = eTagHeader.getValue();
 
         eTag = eTag.replace("\"", "");
-        headers = unprocessedResponse.getHeaders();
+        HttpHeaders headers = unprocessedResponse.getHeaders();
         headers.put(eTagHeader.getName(), eTag);
-        innerHttpResponse = unprocessedResponse;
-        return new InnerHttpResponse(unprocessedResponse.getRequest());
+        return new InnerHttpResponse(unprocessedResponse, headers, unprocessedResponse.getRequest());
     }
 
     private final class InnerHttpResponse extends HttpResponse {
-        protected InnerHttpResponse(HttpRequest request) {
+        private final HttpResponse innerHttpResponse;
```

Comment:
@sima-zhu These are the changes I was discussing with you in Teams. Please let me know if you have concerns about changing the scope of these variables.

Reply:
Sima mentioned offline she is fine with these changes.